### ISSUE TITLE
LuaJIT-tests suite intermittently fails on M1 Macs

### ISSUE THREAD

**Issue by igormunkin** [tarantool/tarantool#6098]
Running `make LuaJIT-tests` ends with the following failure sometimes.

<details>
<summary>LuaJIT-tests failure</summary>

```bash
$ make LuaJIT-tests
[  1%] Built target msgpuck
[  1%] Built target coro
[  1%] Built target eio
[  1%] Built target decNumber
[  3%] Built target ev
[  3%] Built target bit
[  5%] Built target small
[  7%] Built target salad
[  7%] Built target uri
[ 12%] Built target core
[ 14%] Built target stat
[ 14%] Built target uuid
[ 14%] Built target mpstream
[ 14%] Built target vclock
[ 14%] Built target box_error
[ 19%] Built target zstd
[ 19%] Built target cpu_feature
[ 21%] Built target crc32
[ 21%] Built target xlog
[ 21%] Built target misc
[ 23%] Built target bundled-ares-project
[ 23%] Built target bundled-libcurl-project
[ 25%] Built target yaml
[ 26%] Built target minilua
[ 26%] Built target buildvm
[ 28%] Built target buildvm_output
[ 39%] Built target core_static
[ 39%] Built target vm_static
[ 39%] Built target libluajit_static
[ 39%] Built target libluajit
[ 39%] Built target build_bundled_libs
[ 39%] Built target scramble
[ 41%] Built target txt2c
[ 42%] Built target http_parser
[ 44%] Built target coll
[ 44%] Built target crypto
[ 46%] Built target swim_udp
[ 46%] Built target swim_ev
[ 46%] Built target swim
[ 67%] Built target server
[ 67%] Built target shutdown
[ 67%] Built target bitset
[ 67%] Built target csv
[ 67%] Built target json
[ 67%] Built target raft
[ 67%] Built target bin2c
[ 69%] Built target lemon
[ 69%] Built target mkkeywordhash
[ 69%] Built target generate_sql_files
[ 73%] Built target tuple
[ 73%] Built target xrow
[100%] Built target box
[100%] Built target tarantool
Running LuaJIT-tests
[  1/502] lang/andor.lua --- smoke
[  2/502] lang/andor.lua --- allcases
[  3/502] lang/andor.lua --- tracefib
[  4/502] lang/assignment.lua --- local
[  5/502] lang/assignment.lua --- global
[  6/502] lang/assignment.lua --- local lhs in key on lhs
[  7/502] lang/assignment.lua --- global lhs in key on lhs
[  8/502] lang/compare.lua --- 1,2
[  9/502] lang/compare.lua --- 2,1
[ 10/502] lang/compare.lua --- 1,1
[ 11/502] lang/compare.lua --- 2
[ 12/502] lang/compare.lua --- 1
[ 13/502] lang/compare.lua --- 0
[ 14/502] lang/compare.lua --- pcall
[ 15/502] lang/compare.lua --- bit
[ 16/502] lang/compare.lua --- string 1 255
[ 17/502] lang/compare.lua --- String comparisons:
[ 18/502] lang/compare.lua --- obj_eq/ne
[ 19/502] lang/compare_nan.lua --- nan nan
[ 20/502] lang/compare_nan.lua --- nan one
[ 21/502] lang/compare_nan.lua --- one nan
[ 22/502] lang/compare_nan.lua --- nan 1
[ 23/502] lang/compare_nan.lua --- 1 nan
[ 24/502] lang/compare_nan.lua --- not nan nan
[ 25/502] lang/compare_nan.lua --- not nan one
[ 26/502] lang/compare_nan.lua --- not one nan
[ 27/502] lang/compare_nan.lua --- not nan 1
[ 28/502] lang/compare_nan.lua --- not 1 nan
[ 29/502] lang/constant/number.lua --- exp
[ 30/502] lang/constant/number.lua --- hex exp
[ 31/502] lang/constant/table.lua --- tnew
[ 32/502] lang/constant/table.lua --- tdup
[ 33/502] lang/for.lua --- direction
[ 34/502] lang/for.lua --- coerce to integer at 13
[ 35/502] lang/for.lua --- coerce to integer at 10
[ 36/502] lang/for.lua --- cannot coerce to integer at 10
[ 37/502] lang/length.lua --- length increasing and decreasing in loop
[ 38/502] lang/length.lua --- length increasing in loop with existing element
[ 39/502] lang/length.lua --- length decreasing in loop with erased element
[ 40/502] lang/modulo.lua --- integer equivalence
[ 41/502] lang/modulo.lua --- fractional equivalence
[ 42/502] lang/modulo.lua --- jit constant RHS
[ 43/502] lang/modulo.lua --- jit constant LHS, with exit
[ 44/502] lang/modulo.lua --- divide by zero
[ 45/502] lang/concat.lua --- Constant folding
[ 46/502] lang/concat.lua --- Fuse conversions to strings
[ 47/502] lang/concat.lua --- Fuse string construction
[ 48/502] lang/concat.lua --- CSE, sink
[ 49/502] lang/concat.lua --- CSE, two buffers in parallel, no sink
[ 50/502] lang/concat.lua --- Append, CSE
[ 51/502] lang/concat.lua --- Append, FOLD
[ 52/502] lang/concat.lua --- Append to buffer, sink
[ 53/502] lang/concat.lua --- Append to two buffers in parallel, no append, no sink
[ 54/502] lang/concat.lua --- Sink into side-exit
[ 55/502] lang/concat.lua --- Very long strings
[ 56/502] lang/self.lua --- trivial setget
[ 57/502] lang/table.lua --- tables as keys in tables
[ 58/502] lang/table.lua --- some tables as keys in tables
[ 59/502] lang/upvalue/closure.lua --- for
[ 60/502] lang/upvalue/closure.lua --- while
[ 61/502] lang/upvalue/closure.lua --- repeat
[ 62/502] lang/upvalue/closure.lua --- func
[ 63/502] lang/upvalue/closure.lua --- recursive type change
[ 64/502] lang/upvalue/closure.lua --- Don't mark upvalue as immutable if written to after prototype definition
[ 65/502] lang/coroutine.lua --- traceback
[ 66/502] lang/tail_recursion.lua --- self
[ 67/502] lang/tail_recursion.lua --- mutual
[ 68/502] lang/vararg_jit.lua --- 1
[ 69/502] lang/vararg_jit.lua --- 2
[ 70/502] lang/vararg_jit.lua --- 3
[ 71/502] lang/vararg_jit.lua --- 4
[ 72/502] lang/vararg_jit.lua --- 5
[ 73/502] lang/vararg_jit.lua --- 6
[ 74/502] lang/vararg_jit.lua --- 7
[ 75/502] lang/gc.lua --- rechain
[ 76/502] lang/gc.lua --- TSETM gc
[ 77/502] lang/goto.lua --- Basic goto and label semantics.
[ 78/502] lang/goto.lua --- Goto is not a keyword.
[ 79/502] lang/goto.lua --- Trailing label is considered to be out of scope.
[ 80/502] lang/goto.lua --- Simple loop with cross-jumping.
[ 81/502] lang/goto.lua --- Backwards goto must close upval.
[ 82/502] lang/goto.lua --- Break must close upval, even if closure is parsed after break.
[ 83/502] lang/goto.lua --- Label prevents joining to KNIL.
[ 84/502] lang/goto.lua --- Break resolved from the right scope.
[ 85/502] lang/goto.lua --- Do not join twice with UCLO.
[ 86/502] lang/meta/arith.lua --- op
[ 87/502] lang/meta/arith.lua --- lhs
[ 88/502] lang/meta/arith.lua --- rhs
[ 89/502] lang/meta/arith.lua --- meta only lhs
[ 90/502] lang/meta/arith.lua --- meta only rhs
[ 91/502] lang/meta/arith.lua --- defaults string, int
[ 92/502] lang/meta/arith.lua --- defaults int, string
[ 93/502] lang/meta/arith.lua --- defaults string, string
[ 94/502] lang/meta/arith.lua --- defaults string, kint
[ 95/502] lang/meta/arith.lua --- defaults kint, string
[ 96/502] lang/meta/arith_jit.lua --- assert rhs
[ 97/502] lang/meta/arith_jit.lua --- assert lhs
[ 98/502] lang/meta/arith_jit.lua --- assert both sides
[ 99/502] lang/meta/arith_jit.lua --- adjust no result to one result
[100/502] lang/meta/call.lua --- table
[101/502] lang/meta/call.lua --- userdata
[102/502] lang/meta/call.lua --- number
[103/502] lang/meta/call.lua --- table with changing metamethod
[104/502] lang/meta/call.lua --- jit table
[105/502] lang/meta/call.lua --- jit table rawget as metamethod
[106/502] lang/meta/call.lua --- jit number
[107/502] lang/meta/call.lua --- jit newindex pcall
[108/502] lang/meta/call.lua --- jit index pcall
[109/502] lang/meta/cat.lua --- default
[110/502] lang/meta/cat.lua --- lhs
[111/502] lang/meta/cat.lua --- rhs
[112/502] lang/meta/cat.lua --- mixed types
[113/502] lang/meta/cat.lua --- jit mixed types
[114/502] lang/meta/comp.lua --- __lt and __le xop
[115/502] lang/meta/comp.lua --- __lt and __le values
[116/502] lang/meta/comp_jit.lua --- coverage
[117/502] lang/meta/comp_jit.lua --- Mixed metamethods for ordered comparisons.
[118/502] lang/meta/eq.lua --- __eq xop
[119/502] lang/meta/eq.lua --- __eq values
[120/502] lang/meta/eq_jit.lua --- coverage
[121/502] lang/meta/eq_jit.lua --- non-constant objects
[122/502] lang/meta/framegap.lua --- untitled
[123/502] lang/meta/index.lua --- table 1
[124/502] lang/meta/index.lua --- table 2
[125/502] lang/meta/index.lua --- userdata
[126/502] lang/meta/index.lua --- string
[127/502] lang/meta/len.lua --- table
[128/502] lang/meta/len.lua --- userdata
[129/502] lang/meta/len.lua --- number
[130/502] lang/meta/newindex.lua --- table 1
[131/502] lang/meta/newindex.lua --- jit gaining href
[132/502] lang/meta/newindex.lua --- jit gaining aref
[133/502] lang/meta/newindex.lua --- resize
[134/502] lang/meta/newindex.lua --- str
[135/502] lang/meta/nomm.lua --- untitled
[136/502] lang/meta/debuginfo.lua --- table metamethods
[137/502] lang/meta/debuginfo.lua --- userdata metamethods
[138/502] lib/base/assert.lua --- pass through one
[139/502] lib/base/assert.lua --- pass through many
[140/502] lib/base/assert.lua --- raise on nil
[141/502] lib/base/assert.lua --- raise on false
[142/502] lib/base/error.lua --- no message
[143/502] lib/base/error.lua --- level 0
[144/502] lib/base/error.lua --- default level
[145/502] lib/base/error.lua --- default level in xpcall
[146/502] lib/base/error.lua --- level 2 in xpcall
[147/502] lib/base/getfenv.lua --- untitled
[148/502] lib/base/getsetmetatable.lua --- get __metatable
[149/502] lib/base/getsetmetatable.lua --- jit smoke
[150/502] lib/base/getsetmetatable.lua --- jit assorted
[151/502] lib/base/getsetmetatable.lua --- jit get primitive metatable
[152/502] lib/base/ipairs.lua --- small integer values
[153/502] lib/base/ipairs.lua --- jit key=value
[154/502] lib/base/ipairs.lua --- untitled
[155/502] lib/base/next.lua --- _G 1
[156/502] lib/base/next.lua --- as iterator
[157/502] lib/base/pairs.lua --- nometatable
[158/502] lib/base/pairs.lua --- empty metatable
[159/502] lib/base/pairs.lua --- _G
[160/502] lib/base/pairs.lua --- count
[161/502] lib/base/pcall_jit.lua --- square sum
[162/502] lib/base/pcall_jit.lua --- sqrt square sum
[163/502] lib/base/pcall_jit.lua --- sum with error
[164/502] lib/base/pcall_jit.lua --- sum or square
[165/502] lib/base/pcall_jit.lua --- sum or square with error
[166/502] lib/base/select.lua --- select #
[167/502] lib/base/select.lua --- select modf
[168/502] lib/base/select.lua --- select 1 
[169/502] lib/base/select.lua --- select 2
[170/502] lib/base/select.lua --- select vararg #
[171/502] lib/base/select.lua --- select vararg i
[172/502] lib/base/select.lua --- select vararg 4
[173/502] lib/base/select.lua --- varg-select specialisation requires guard against select
[174/502] lib/base/tonumber_tostring.lua --- tonumber int
[175/502] lib/base/tonumber_tostring.lua --- tonumber float
[176/502] lib/base/tonumber_tostring.lua --- tostring int / tonumber
[177/502] lib/base/tonumber_tostring.lua --- tostring float / tonumber
[178/502] lib/base/tonumber_tostring.lua --- tonumber table
[179/502] lib/base/tonumber_tostring.lua --- tostring int / tostring
[180/502] lib/base/tonumber_tostring.lua --- tostring table __tostring
[181/502] lib/base/tonumber_tostring.lua --- tostring table __tostring __call
[182/502] lib/base/tonumber_tostring.lua --- print calls overridden tostring
[183/502] lib/base/tonumber_tostring.lua --- tonumber base 2
[184/502] lib/base/tonumber_tostring.lua --- __tostring must be callable
[185/502] lib/bit.lua --- Minimal sanity checks.
[186/502] lib/bit.lua --- Basic argument processing.
[187/502] lib/bit.lua --- unop test vectors
[188/502] lib/bit.lua --- binop test vectors
[189/502] lib/bit.lua --- shift test vectors
[190/502] lib/bit.lua --- tohex test vectors
[191/502] lib/bit.lua --- Don't propagate TOBIT narrowing across two conversions.
[192/502] lib/math/abs.lua --- smoke
[193/502] lib/math/abs.lua --- argcheck
[194/502] lib/math/constants.lua --- pi
[195/502] lib/math/constants.lua --- huge
[196/502] lib/math/random.lua --- generally uniform in range [0, 1)
[197/502] lib/math/random.lua --- all in range [1, 10]
[198/502] lib/math/random.lua --- all in range [-3, 11]
[199/502] lib/string/metatable.lua --- __index metamethod is string library
[200/502] lib/string/byte.lua --- simple
[201/502] lib/string/byte.lua --- Fixed slice [i,i+k] or overflow
[202/502] lib/string/byte.lua --- Positive slice [i,len] or overflow
[203/502] lib/string/byte.lua --- Negative slice [-i,len] or underflow
[204/502] lib/string/byte.lua --- Positive slice [1,i] or overflow
[205/502] lib/string/byte.lua --- Negative slice [1,-i] or underflow
[206/502] lib/string/byte.lua --- Check for slot stack overflow
[207/502] lib/string/char.lua --- jit one char
[208/502] lib/string/char.lua --- jit until out of bounds
[209/502] lib/string/char.lua --- jit five chars
[210/502] lib/string/dump.lua --- Must unpatch modified bytecode with ILOOP/JLOOP etc.
[211/502] lib/string/dump.lua --- roundtrip constants
[212/502] lib/string/format/num.lua --- small denormals at low precision
[213/502] lib/string/format/num.lua --- smoke
[214/502] lib/string/format/num.lua --- easily enumerable cases of %a, %A
[215/502] lib/string/format/num.lua --- easily enumerable cases of %f
[216/502] lib/string/format/num.lua --- easily enumerable cases of %e
[217/502] lib/string/format/num.lua --- assorted
[218/502] lib/string/format/num.lua --- assorted %a
[219/502] lib/string/format/num.lua --- Cases where inprecision can easily affect rounding
[220/502] lib/string/format/num.lua --- ExploringBinary.com/print-precision-of-dyadic-fractions-varies-by-language/
[221/502] lib/string/format/num.lua --- ExploringBinary.com/incorrect-floating-point-to-decimal-conversions/
[222/502] lib/string/format/num.lua --- big f
[223/502] lib/string/format/num.lua --- RandomASCII.wordpress.com/2013/02/07/
[224/502] lib/string/len.lua --- smoke
[225/502] lib/string/len.lua --- argcheck
[226/502] lib/string/lower_upper.lua --- smoke
[227/502] lib/string/lower_upper.lua --- repeated
[228/502] lib/string/lower_upper.lua --- repeated with growing string
[229/502] lib/string/lower_upper.lua --- misc upper
[230/502] lib/string/lower_upper.lua --- misc lower
[231/502] lib/string/multiple_functions.lua --- string_op
[232/502] lib/string/rep.lua --- smoke
[233/502] lib/string/rep.lua --- versus concat
[234/502] lib/string/rep.lua --- misc
[235/502] lib/string/rep.lua --- separator
[236/502] lib/string/rep.lua --- iterate to table
[237/502] lib/string/rep.lua --- iterate to table with sep
[238/502] lib/string/rep.lua --- iterate and concat
[239/502] lib/string/reverse.lua --- misc
[240/502] lib/string/sub.lua --- smoke
[241/502] lib/string/sub.lua --- argcheck
[242/502] lib/string/sub.lua --- all bar substrings
[243/502] lib/string/sub.lua --- Positive slice [i,len] or overflow
[244/502] lib/string/sub.lua --- Negative slice [-i,len] or underflow
[245/502] lib/string/sub.lua --- Positive slice [1,i] or overflow
[246/502] lib/string/sub.lua --- Negative slice [1,-i] or underflow
[247/502] lib/string/sub.lua --- jit sub 1 eq
[248/502] lib/string/sub.lua --- jit sub 1 ne (contents)
[249/502] lib/string/sub.lua --- jit sub 1 ne (rhs too long)
[250/502] lib/string/sub.lua --- jit sub 1,2 ne
[251/502] lib/string/sub.lua --- jit sub 1,k eq
[252/502] lib/string/sub.lua --- jit sub 1,k ne (contents)
[253/502] lib/string/sub.lua --- jit sub 1,k ne (rhs too long)
[254/502] lib/string/sub.lua --- jit sub 1,2 eq
[255/502] lib/string/sub.lua --- jit sub 1,3 eq
[256/502] lib/string/sub.lua --- jit sub 1,4 eq
[257/502] lib/string/sub.lua --- jit sub i,i
[258/502] lib/table/concat.lua --- table.concat
[259/502] lib/table/concat.lua --- table.concat must inhibit CSE and DSE
[260/502] lib/table/concat.lua --- table.concat must inhibit CSE and DSE 2
[261/502] lib/table/concat.lua --- table.concat must inhibit CSE and DSE 3
[262/502] lib/table/insert.lua --- table.insert(t,i)
[263/502] lib/table/insert.lua --- table.insert(t,i,i)
[264/502] lib/table/new.lua --- table.new
[265/502] lib/table/remove.lua --- table.remove(t) removes correct entries
[266/502] lib/table/remove.lua --- table.remove(t) returns the removed entry
[267/502] lib/table/remove.lua --- table.remove(t, 1) removes and returns the first entry
[268/502] lib/table/remove.lua --- TSETR hash part
[269/502] lib/table/remove.lua --- TSETR write barrier
[270/502] lib/table/sort.lua --- numbers
[271/502] lib/table/sort.lua --- strings
[272/502] lib/table/sort.lua --- tables
[273/502] lib/coroutine/yield.lua --- Stack overflow on return (create)
[274/502] lib/coroutine/yield.lua --- Stack overflow on return (wrap)
[275/502] lib/coroutine/yield.lua --- cogen
[276/502] lib/coroutine/yield.lua --- cofunc
[277/502] lib/coroutine/yield.lua --- assorted
[278/502] lib/ffi/bit64.lua --- smoke tobit
[279/502] lib/ffi/bit64.lua --- smoke band
[280/502] lib/ffi/bit64.lua --- smoke shl
[281/502] lib/ffi/bit64.lua --- smoke tohex
[282/502] lib/ffi/bit64.lua --- tobit/band assorted C types
[283/502] lib/ffi/bit64.lua --- tobit/band negative unsigned enum
[284/502] lib/ffi/bit64.lua --- jit band/bor/bxor
[285/502] lib/ffi/bit64.lua --- jit shift/xor
[286/502] lib/ffi/bit64.lua --- jit rotate/xor
[287/502] lib/ffi/bit64.lua --- jit all ops
[288/502] lib/ffi/cdata_var.lua --- byte array allocations
[289/502] lib/ffi/cdata_var.lua --- int array allocations
[290/502] lib/ffi/copy_fill.lua --- misc
[291/502] lib/ffi/copy_fill.lua --- jit char[10]
[292/502] lib/ffi/copy_fill.lua --- jit char[1]
[293/502] lib/ffi/err.lua --- error in FFI metamethod: don't print metamethod frame.
[294/502] lib/ffi/err.lua --- tailcall in regular metamethod: keep metamethod frame.
[295/502] lib/ffi/err.lua --- error in FFI metamethod: set correct PC.
[296/502] lib/ffi/istype.lua --- 1
[297/502] lib/ffi/istype.lua --- 2
[298/502] lib/ffi/istype.lua --- 3
[299/502] lib/ffi/istype.lua --- 4
[300/502] lib/ffi/istype.lua --- 5
[301/502] lib/ffi/jit_array.lua --- smoke
[302/502] lib/ffi/jit_array.lua --- int array pointer arithmetic
[303/502] lib/ffi/jit_array.lua --- double array pointer arithmetic
[304/502] lib/ffi/jit_array.lua --- double array pointer comparisons
[305/502] lib/ffi/jit_array.lua --- constant offset in double array index
[306/502] lib/ffi/jit_array.lua --- fixed index of minus one
[307/502] lib/ffi/jit_array.lua --- uint8_t array element comparisons
[308/502] lib/ffi/jit_array.lua --- int32_t array bit/bswap tricks
[309/502] lib/ffi/jit_array.lua --- int32_t shift/rotate/and
[310/502] lib/ffi/jit_complex.lua --- field access
[311/502] lib/ffi/jit_complex.lua --- one element circular chain, named indexing
[312/502] lib/f/bin/sh: line 1: 45194 Bus error: 10           /Users/tntmac07.tarantool.i/imun/tarantool/src/tarantool -e dofile[[/Users/tntmac07.tarantool.i/imun/tarantool/test/luajit-test-init.lua]] /Users/tntmac07.tarantool.i/imun/tarantool/third_party/luajit/test/LuaJIT-tests/test.lua +slow +ffi +bit +jit
make[3]: *** [LuaJIT-tests] Error 138
make[2]: *** [third_party/luajit/test/LuaJIT-tests/CMakeFiles/LuaJIT-tests.dir/all] Error 2
make[1]: *** [third_party/luajit/test/LuaJIT-tests/CMakeFiles/LuaJIT-tests.dir/rule] Error 2
make: *** [LuaJIT-tests] Error 2
```

</details>

**Comment by igormunkin** [comment]
The issue is resolved in scope of tarantool/luajit@814625f. The submodule was updated in tarantool in 2.9.0-345-g847b096e3 (847b096).
